.github/workflows/release.yml

```diff
@@ -0,0 +1,41 @@
+# This is a GitHub workflow to auto-deploy releases on tags
+# Useful references:
+# https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
+# https://docs.github.com/en/free-pro-team@latest/actions/guides/building-and-testing-python#publishing-to-package-registries
+# https://github.community/t/how-to-run-github-actions-workflow-only-for-new-tags/16075
+# https://www.codingwithcalvin.net/git-tag-based-released-process-using-github-actions/
+name: Release
+
+# Trigger the workflows on all PRs pushed commits and/or tags
+# to be able to test changes without opening PRs
+on:
+  push:
+    # FIXME: use the release event instead?
+    tags: [0-9]+.[0-9]+.[0-9]+
+    branches: master
```

Comment:
This will not support any beta releases etc though, its worth making a note about or add support for, for example by appedning .* in the end.